docusaurus/docs/engineering/knowledge-base/development-lifecycle.md

```diff
@@ -4,154 +4,239 @@ sidebar_position: 7
+1. **Add a copy of the project's code to your own GitHub account.**
 
-   For example, if the developer is working on adding a "read more" feature, they might create a branch called `read-more`. Creating branches allows a developer to work on many independent features at the same time.
+   > This first step involves getting your own version of the project's code to edit to your heart's content. In technical terms, this means creating a 'fork' of the project you want to contribute to.
 
-2. **The developer "commits" code changes to their "feature branch".** As the developer writes code, they will modify the code in the branch they're working on. Periodically, the developer will then "commit" the changes made. Committing creates "code checkpoints", and is helpful to split code creation in logical segment that are easier to reason about and check for correctness.
+1. **Download the code (from your GitHub account) to your computer.**
 
-   For example, when working on the `read-more` branch, the developer might create the following commits:
+   > You need to get the code onto your computer to start making changes. This way you can edit the code in the comfort of your own ~~home~~ code editor. This is done by downloading or, in technical terms, 'cloning' the code from your GitHub account.
 
-   - Commit 1: Add read more button
-   - Commit 2: Style read more button
-   - Commit 2: Display read more text when button is clicked
+1. **Create a Branch in the code to keep your changes contained.**
 
-3. **The developer opens a "pull request" on the original codebase.** Once the new feature is complete, it should become part of the "official" codebase. To do so, the developer must ask the codebase owners to copy the commits made in step 2 to the `master` branch of the original `upstream` codebase. This process is called opening a "pull request".
-4. **The codebase owners review (and eventually "merge") the pull request.** Once the pull request has been submitted for review, the codebase owners will review the changes made and might request edits to the code. The review process usually entails multiple rounds of back and forth between the developer and the codebase owners, and the developer often adds new commits to the pull request with the requested changes.
+   > You get to create parallel versions of the code so you can work on different unrelated things at the same time without getting them tangled.
 
-   Once the pull request is approved, the pull request is "merged" and then "closed". Once merged, the developer's code becomes part of the `master` branch in the original `upstream` codebase. The developer can now delete the "feature branch" created in step 1.
+1. **Make changes and set checkpoints as you get small tasks done.**
```

Comment:
I think "mark checkpoints" reads better than "set checkpoints". When I read set checkpoints I imagined a debugger.

I would also say:

> It's always good to break up any tasks into smaller steps. Regularly marking checkpoints as you accomplish small tasks will let you see all the steps you took along the way, and course-correct in case of issues. In technical terms, marking checkpoints by logging changes with descriptive messages in git is referred to as 'making commits'.